From 4bd8cb8989ffb674cc9c64ef3903ea081292f63d Mon Sep 17 00:00:00 2001
From: HashBang <hashbang173@gmail.com>
Date: Mon, 1 Aug 2016 14:36:46 -0400
Subject: [PATCH 17/39] extract_utils: Do not change app SRC if there are
 arguments

Do not prepend app or priv-app to the src when a path is explicitly
given as an argument. This fixes the SRC path in such conditions.

Change-Id: I43370f17fe224df323fcfa9b0d5eae4ee2996524
---
 build/tools/extract_utils.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index b883219f..13098a0b 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -255,10 +255,12 @@ function write_packages() {
                 printf 'LOCAL_MULTILIB := %s\n' "$EXTRA"
             fi
         elif [ "$CLASS" = "APPS" ]; then
-            if [ "$EXTRA" = "priv-app" ]; then
-                SRC="$SRC/priv-app"
-            else
-                SRC="$SRC/app"
+            if [ -z "$ARGS" ]; then
+                if [ "$EXTRA" = "priv-app" ]; then
+                    SRC="$SRC/priv-app"
+                else
+                    SRC="$SRC/app"
+                fi
             fi
             printf 'LOCAL_SRC_FILES := %s/%s\n' "$SRC" "$FILE"
             local CERT=platform
-- 
2.15.0

